quic/s2n-quic-transport/src/space/datagram.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    endpoint,
+    stream::{AbstractStreamManager, StreamTrait as Stream},
+    transmission::WriteContext,
+};
+use s2n_quic_core::{
+    datagram::{Endpoint, Sender},
+    frame,
+};
+
+pub struct Manager<Config: endpoint::Config> {
+    sender: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Sender,
+    // TODO: Remove this warning once Receiver is implemented
+    #[allow(dead_code)]
+    receiver: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Receiver,
+}
+
+impl<Config: endpoint::Config> Manager<Config> {
+    pub fn new(
+        sender: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Sender,
+        receiver: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Receiver,
+    ) -> Self {
+        Self { sender, receiver }
+    }
+
+    pub fn on_transmit<S: Stream, W: WriteContext>(
```

Comment:
can you add some docs here.

Reply:
When you say docs, do you mean for the user or for us?

Reply:
So for the `Sending` and `Receiver` trait you might want docs geared towards the user. For internal structures the docs would be for us.

Mainly asking for docs now since you are implementing it now and you know what its doing. Coming back later will be more painful.

So in Rust the documentation for `functions` is formatted as:

```
A quick 1 sentence summary of the function.

More details if necessary. Explain an confusing caveats or interesting behavior
that the function assumes or does.

This could also be some design choice or some future consideration.
```